src/Summoner/Tui/Validation.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE Rank2Types #-}
+
+-- | This module contains function to validate Form fields.
+
+module Summoner.Tui.Validation
+       ( ctrlD
+       , summonFormValidation
+       , fieldNameErrorMsg
+       ) where
+
+import Brick.Focus (focusGetCurrent)
+import Brick.Forms (Form, formFocus, formState, setFieldValid)
+import Lens.Micro (Lens', (.~), (^.))
+
+import Summoner.Tui.Form (SummonForm (..), mkForm)
+import Summoner.Tui.Kit
+
+
+type KitForm e = Form SummonKit e SummonForm
+
+-- | Clears the 'Text' fields by @Ctrl + d@ key combination.
+ctrlD :: KitForm e -> KitForm e
+ctrlD =
+      clearField UserFullName (user . fullName)
+    . clearField UserEmail (user . email)
+    . clearField ProjectName (project . repo)
+    . clearField ProjectDesc (project . desc)
+    . clearField ProjectCat (project . category)
+    . clearField CustomPreludeName (projectMeta . preludeName)
+    . clearField CustomPreludeModule (projectMeta . preludeModule)
+    . clearField UserOwner (user . owner)
+  where
+    clearField :: SummonForm -> Lens' SummonKit Text -> KitForm e -> KitForm e
```

Comment:
After thinking about the way to clean other fields rather than `Text` I have one idea. You can implement function where you pass value to set with lens, like:

```haskell
clearField :: a -> SummonForm -> Lens' SummonKit a -> KitForm e -> KitForm e
```

And then you can have:

```haskell
clearTextField :: SummonForm -> Lens' SummonKit Text -> KitForm e -> KitForm e
clearTextField = clearField ""

clearListField :: SummonForm -> Lens' SummonKit [a] -> KitForm e -> KitForm e
clearListField = clearField []
```

Or you can pass `""` or `[]` directly as arguments (looks like `clearField ""` is much shorter than `clearTextField`). Not sure it worth the complication but maybe useful if we want to be able to clear all fields.